account_consolidation/wizard/consolidation_consolidate.py

```diff
@@ -151,18 +162,52 @@ def reverse_moves(self, subsidiary):
                               Recordset of the reversal moves
         """
         move_obj = self.env['account.move']
-        move_to_reverse = move_obj.search(
+        moves_to_reverse = move_obj.search(
             [('journal_id', '=', self.journal_id.id),
              ('to_be_reversed', '=', True),
-             ('consol_company_id', '=', subsidiary.id)])
-        reversal_action = self.env['account.move.reverse'].with_context(
-            active_ids=move_to_reverse.ids).create({
-                'date': self._get_month_first_date(),
-                'journal_id': self.journal_id.id
-            }).action_reverse()
+             ('consol_company_id', '=', subsidiary.id),
+             ('reversal_id', '=', False)])
+        if not moves_to_reverse:
+            return moves_to_reverse, False
+        try:
+            reversal_action = self.env['account.move.reverse'].with_context(
+                active_ids=moves_to_reverse.ids,
+                __conso_reversal_no_post=True).create({
+                    'date': self._get_month_first_date(),
+                    'journal_id': self.journal_id.id
+                }).action_reverse()
+        except ValidationError as e:
+            raise ValidationError(_(
+                "The error below appeared while trying to reverse the "
+                "following moves: \n %s \n %s") % (
+                '\n'.join(['- %s' % m.name for m in moves_to_reverse]),
+                e.name
+            ))
+        # Restore flag to be reversed after it was removed by action_reverse
+        # We need this to make sure the move will be reversed again on the next
+        # run if the reversal move was deleted
+        # As reversed move have the reversal_id link, and we select only the
+        # moves not having it in the search above, we can be sure that only the
+        # last consolidation moves will be reversed
+        moves_to_reverse.write({'to_be_reversed': True})
```

Comment:
is this still the case? shouldn't we relay only on reversal_id in search, or we should also override unlink to raise flag